dc/s2n-quic-dc/events/connection.rs

```diff
@@ -1,26 +1,209 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#[event("application:write")]
-pub struct ApplicationWrite {
+#[event("stream:write_flushed")]
+pub struct StreamWriteFlushed {
     /// The number of bytes that the application tried to write
     #[measure("provided", Bytes)]
-    total_len: usize,
+    provided_len: usize,
 
     /// The amount that was written
     #[measure("committed", Bytes)]
     #[counter("committed.total", Bytes)]
-    write_len: usize,
+    committed_len: usize,
+
+    /// Indicates that the write was the final offset of the stream
+    is_fin: bool,
+
+    /// The amount of time it took to process the write request
+    ///
+    /// Note that this includes both any syscall and encryption overhead
+    #[measure("processing_duration", Duration)]
```

Comment:
I guess this is OK for now, but I wonder if we should have a duration_per_byte or something like that, to make it easier to actually make sense of what this means.

Reply:
yeah that could be an interesting metric to have. i can add that as a follow-up